rosbag2_storage_default_plugins/include/rosbag2_storage_default_plugins/sqlite/sqlite_storage.hpp

```diff
@@ -64,6 +64,8 @@ class ROSBAG2_STORAGE_DEFAULT_PLUGINS_PUBLIC SqliteStorage
 
   rosbag2_storage::BagMetadata get_metadata() override;
 
+  std::string get_relative_path() const override;
```

Comment:
Thinking about this, I believe `get_metadata()` should not be part of each storage interface as it is actually storage agnostic. That being said, `get_relative_path()` should IMO just be renamed to `get_current_file_path()` or similar. The metadata should then be solely handled in the writer/reader level. 
Otherwise, I feel that `get_relative_path()` is in conflict with `get_metadata()` as the latter should be returning a struct with all the relative paths set to it.

Reply:
I agree here. This PR and #184 are moving ownership of metadata out of the storage plugin and into `Writer`. 

The reason it is `get_relative_path()` instead of `get_current_file_path` or `get_relative_paths` is because only one path is opened by the storage plugin when the Writer invokes `open_read_write`. In the event of bagfile splitting, `Writer` calls `open_read_write` again once the bagfile should split, but with the next relative file path.